src/generic/ondemand/json_iterator-inl.h

```diff
@@ -0,0 +1,329 @@
+  logger::log_value(*this, "uint64", "", 0);
+  return stage2::numberparsing::parse_unsigned(buf);
+}
+SIMDJSON_WARN_UNUSED simdjson_result<int64_t> json_iterator::get_root_int64() noexcept {
+  uint8_t tmpbuf[20+1]; // -<19 digits> is the longest possible integer 
+  if (!advance_to_buffer(tmpbuf)) { return NUMBER_ERROR; }
+  logger::log_value(*this, "int64", "", 0);
+  return stage2::numberparsing::parse_integer(buf);
+}
+SIMDJSON_WARN_UNUSED simdjson_result<double> json_iterator::get_root_double() noexcept {
+  // Per https://www.exploringbinary.com/maximum-number-of-decimal-digits-in-binary-floating-point-numbers/, 1074 is the maximum number of significant fractional digits. Add 8 more digits for the biggest number: -0.<fraction>e-308.
+  uint8_t tmpbuf[1074+8+1];
+  if (!advance_to_buffer(tmpbuf)) { return NUMBER_ERROR; }
+  logger::log_value(*this, "double", "", 0);
+  return stage2::numberparsing::parse_double(buf);
+}
+SIMDJSON_WARN_UNUSED simdjson_result<bool> json_iterator::get_root_bool() noexcept {
+  uint8_t tmpbuf[5+1];
+  if (!advance_to_buffer(tmpbuf)) { return INCORRECT_TYPE; } // Too big! Can't be true or false
+  return get_bool();
+}
+simdjson_really_inline bool json_iterator::root_is_null() noexcept {
+  uint8_t tmpbuf[4+1];
+  if (!advance_to_buffer(tmpbuf)) { return false; } // Too big! Can't be null
+  return is_null();
+}
+
+SIMDJSON_WARN_UNUSED simdjson_really_inline error_code json_iterator::skip() noexcept {
+  switch (*advance()) {
+    // PERF TODO does it skip the depth check when we don't decrement depth?
+    case '[': case '{':
+      logger::log_start_value(*this, "skip");
+      return skip_container();
+    default:
+      logger::log_value(*this, "skip", "");
+      return SUCCESS;
+  }
+}
+
+SIMDJSON_WARN_UNUSED simdjson_really_inline error_code json_iterator::skip_container() noexcept {
+  uint32_t depth = 1;
+  // The loop breaks only when depth-- happens.
+  auto end = &parser->dom_parser.structural_indexes[parser->dom_parser.n_structural_indexes];
+  while (index <= end) {
+    uint8_t ch = *advance();
+    switch (ch) {
+      // TODO consider whether matching braces is a requirement: if non-matching braces indicates
+      // *missing* braces, then future lookups are not in the object/arrays they think they are,
+      // violating the rule "validate enough structure that the user can be confident they are
+      // looking at the right values."
+      case ']': case '}':
+        logger::log_end_value(*this, "skip");
+        depth--;
+        if (depth == 0) { logger::log_event(*this, "end skip", ""); return SUCCESS; }
+        break;
+      // PERF TODO does it skip the depth check when we don't decrement depth?
+      case '[': case '{':
+        logger::log_start_value(*this, "skip");
+        depth++;
+        break;
+      default:
+        logger::log_value(*this, "skip", "");
+        break;
+    }
+  }
+
+  logger::log_error(*this, "not enough close braces");
```

Comment:
close braces => closing braces ?

Reply:
Sure. Will need to change open to opening everywhere too.